hyperspy/tests/component/test_component_active_array.py

```diff
@@ -52,15 +53,14 @@ def test_toggle_array(self):
         g._create_arrays()
         g._active_array[2, 0] = False
         g._active_array[0, 0] = False
-        g._toggle_connect_active_array(False)
-        nt.assert_false(g._active_is_multidimensional)
-        nt.assert_true(
-            np.all(
-                g.A.as_signal('values').data == np.zeros(
-                    (3, 3))))
-        nt.assert_true(np.all(
-            g._active_array == np.array([[0, 1, 1], [1, 1, 1], [0, 1, 1]], dtype=bool)))
-        g._toggle_connect_active_array(True)
+        with stash_active_state([g]):
+            g.active_is_multidimensional = False
+            nt.assert_false(g._active_is_multidimensional)
+            nt.assert_true(
+                np.all(
+                    g.A.as_signal('values').data == np.zeros(
+                        (3, 3))))
+            nt.assert_equal(g._active_array, None)
```

Comment:
Ah, now I understand what you were talking about. The test line in question confused me a bit. It explicitly tested the state of the `_active_array` while disconnected. The current PR changes the internal behavior so that the array is set to `None` while stashed. I have re-introduced the test in question on the outside of the `with` block though, as you initially proposed :)